parcels_benchmarks/benchmark_setup.py

```diff
@@ -0,0 +1,264 @@
+    )
+
+def download_example_dataset(dataset: str, data_home=None):
+    """Load an example dataset from the parcels website.
+
+    This function provides quick access to a small number of example datasets
+    that are useful in documentation and testing in parcels.
+
+    Parameters
+    ----------
+    dataset : str
+        Name of the dataset to load.
+    data_home : pathlike, optional
+        The directory in which to cache data. If not specified, the value
+        of the ``PARCELS_EXAMPLE_DATA`` environment variable, if any, is used.
+        Otherwise the default location is assigned by :func:`get_data_home`.
+
+    Returns
+    -------
+    dataset_folder : Path
+        Path to the folder containing the downloaded dataset files.
+    """
+    # Dev note: `dataset` is assumed to be a folder name with netcdf files
+    if dataset not in DATA_FILES:
+        raise ValueError(
+            f"Dataset {dataset!r} not found. Available datasets are: " + ", ".join(DATA_FILES.keys())
+        )
+    odie = _get_pooch(data_home=data_home)
+    print(f"dataset: {dataset}")
+    print(f"Fetching: {DATA_FILES[dataset]}")
+    listing = odie.fetch(DATA_FILES[dataset],processor=pooch.Unzip())
+
+    # as pooch currently returns a file listing while we want a dir,
+    # we find the common parent dir to all files
+    common_parent_dir = min([Path(f) for f in listing], key=lambda f: len(f.parents)).parent
+
+    return common_parent_dir
+
+def download_datasets(data_home=None):
+    """Download all datasets listed in the config file to the specified location.
+
+    Parameters
+    ----------
+    data_home : pathlike, optional
+        The directory in which to cache data. If not specified, the value
+        of the ``PARCELS_EXAMPLE_DATA`` environment variable, if any, is used.
+        Otherwise the default location is assigned by :func:`get_data_home`.
+
+    Returns
+    -------
+    dataset_folders : dict
+        Mapping of dataset names to paths to the folder containing the downloaded dataset files.
+    """
+    dataset_folders = {}
+    for dataset in DATA_FILES:
+        folder = download_example_dataset(dataset, data_home=data_home)
+        dataset_folders[dataset] = folder
+    return dataset_folders
+
+
+def retrieve_data_dir(
+    url: str = None,
+    known_hash: str = None,
+):
+    # let pooch retrieve and get a listing of all included files
+    listing = pooch.retrieve(
+        url,
+        processor=pooch.Unzip(),
+        known_hash=known_hash,
+    )
+
+    # as pooch currently returns a file listing while we want a dir,
+    # we find the common parent dir to all files
+    common_parent_dir = min([Path(f) for f in listing], key=lambda f: len(f.parents)).parent
+
+    return str(common_parent_dir)
+
+
+def list_datasets():
+    """Print available dataset names and their files."""
+    for name, files in DATA_FILES.items():
+        print(f"{name}:")
+        for f in files:
+            print(f"  - {f}")
```

Comment:
Currently, the .values() of DATA_FILES only contain single file names.

```suggestion
    for name, file in DATA_FILES.items():
        print(f"{name}: {file}")
```